druid-shell/src/dialog.rs

```diff
@@ -23,6 +23,8 @@ pub struct FileInfo {
 }
 
 /// Type of file dialog.
+// TODO(x11/file_dialogs): Remove this #[cfg]
+#[cfg(not(all(target_os = "linux", feature = "x11")))]
```

Comment:
does this matter? do things complain if this exists but is unused?

Reply:
Clippy does, yep. Could make it an `#[allow(dead_code)]` like the above instead?

Reply:
I'm confused, if it's a public type clippy shouldn't complain? is it not exposed anywhere? Is there some place where it's used on other platforms but which isn't implemented here?

Reply:
Seems to work now without it, not sure what the problem was before. Removed.

Reply:
Oops, I lied. Failing clippy as of the last commit

Reply:
Fixed.